packages/mobile_app/lib/components/prophecy.dart

```diff
@@ -6,65 +6,92 @@ import 'package:prophecy/bloc.dart';
 import 'package:prophecy_model/prophecy_model.dart';
 import 'prophecy_record.dart';
 import 'prophecy_is_loading.dart';
-import 'package:app/screens/loading.dart';
 import 'package:flutter_svg/flutter_svg.dart';
 import 'package:mutable_wrappers/mutable_wrappers.dart';
 import 'package:users_repository/users_repository.dart';
 import 'package:language/language.dart';
 
 import 'package:int_datetime/int_datetime.dart';
-import 'package:algorithm/astro.dart' show BasicAstrology;
+import 'package:algorithm/algorithm.dart';
 
 class Prophecy extends StatelessWidget {
+  final int dt;
   final UserEntity user;
   final labelStr;
   final birthRow = MutableObject(null);
-  Prophecy({@required this.user})
+  Prophecy({@required this.user, @required this.dt})
       : labelStr = "${user.model.name.capitalize()} (${userRole(user.role)})" {
-    final dt = user.model.birth.toDateTime;
+    //
+    final dtConverted = user.model.birth.toDateTime;
     final sign = user.model.birth.astroSign;
-    final mainPlanet = user.model.birth.astroHousePlanet;
+    //
     birthRow.wrapped = Row(
       children: <Widget>[
         SvgPicture.asset("assets/icons/$sign.svg"),
-        Text(" ${dt.day}.${dt.month}.${dt.year} ",
+        Text(" ${dtConverted.day}.${dtConverted.month}.${dtConverted.year} ",
             style: TextStyle(fontSize: 14)),
-        SvgPicture.asset("assets/icons/$mainPlanet.svg"),
       ],
     );
   }
 
   @override
   Widget build(BuildContext context) {
+    final sign = user.model.birth.astroSign;
+    final userPatron = user.model.birth.astroHousePlanet;
+    final currentPlanets = planetFor[dt.astroSign][sign];
```

Comment:
I'm confused. I see `final int dt;` and here `dt` seem to be an object. How come? :)

Reply:
It is also temporary code I saved =)

Reply:
> I'm confused. I see `final int dt;` and here `dt` seem to be an object. How come? :)

Everything is object, extension allow to convert dt to String of astrologic sign by this date.